### PR TITLE
A mixed bag of updates

### PR DESCRIPTION
**Pull request by ltratt**
This is a bit of a mixed bag of commits -- sorry :/

Performance improvements: https://github.com/softdevteam/error_recovery_experiment/commit/45caadc9a838f814cc2c1070126c9926a0d140e5  https://github.com/softdevteam/error_recovery_experiment/commit/95cbae5900dd46902efe65bdcf167fdf017850e5   https://github.com/softdevteam/error_recovery_experiment/commit/e67927eb9f26bfb805210f0041f3cfb58a8c6b6d  https://github.com/softdevteam/error_recovery_experiment/commit/0828934d50e55c58606f716188a5d0ed0ff79f1d

Process-the-data-better commit: https://github.com/softdevteam/error_recovery_experiment/commit/446e1d3c9d4003d7db018827c5ccdb360fc4bffa

I'm-an-idiot commit: https://github.com/softdevteam/error_recovery_experiment/commit/70e9b7e8695ec487a0ed2abdf31c36f49204c47d

There's-an-annoying-problem-with-our-dataset-that-this-warning-helps-me-spot-and-fix commit: https://github.com/softdevteam/error_recovery_experiment/commit/08c6d9f33df577f692de9a769aecf65c5904db70

On the positive side, they're all pretty tiny.

### CODE REVIEW

File: runner/process.py [hunk 1]
```diff
@@ -255,7 +255,11 @@ def bins_errs(run, num_bins, max_error_locs):
         bbins = [[] for _ in range(num_bins)]
         bin_width = max_error_locs / num_bins
         for _ in range(BOOTSTRAP):
-            d = [len(random.choice(pexecs).costs) for pexecs in run.pexecs]
```
Comment (ptersilie) [inline runner/process.py:249]: Could also be
`d = [len(random.choice(pexecs).costs) for pexecs in run.pexecs if pexec.succeeded]`

Comment (ptersilie) [reply runner/process.py:249]: Ah no, mistook pexec for pexecs. Ignore this.

**Comment by ptersilie** [comment]
The commit message of 446e1d3c9d4003d7db018827c5ccdb360fc4bffa needs rephrasing (probably just a missing word). Rest looks good.

**Comment by ltratt** [comment]
OK, fix pushed. [it's now https://github.com/softdevteam/error_recovery_experiment/pull/13/commits/43f0339471f1dae8f956deecd5c48ccef823dac7]

**Review by ptersilie — APPROVED** [review]

**Pull request merged**